l10n_it_fiscalcode/wizard/compute_fc.py

```diff
@@ -1,54 +1,97 @@
 
 
 class WizardComputeFc(models.TransientModel):
-
-    _name = "wizard.compute.fc"
+    _name = 'wizard.compute.fc'
     _description = "Compute Fiscal Code"
     _rec_name = 'fiscalcode_surname'
 
-    fiscalcode_surname = fields.Char('Surname', size=64)
-    fiscalcode_firstname = fields.Char('First name', size=64)
-    birth_date = fields.Date('Date of birth')
-    birth_city = fields.Many2one(
-        'res.city.it.code.distinct', string='City of birth')
-    birth_province = fields.Many2one(
-        'res.city.it.code.province', string='Province')
-    sex = fields.Selection([
-        ('M', 'Male'),
-        ('F', 'Female'),
-        ], "Sex")
+    fiscalcode_surname = fields.Char("Surname", required=True, size=64)
+    fiscalcode_firstname = fields.Char("First name", required=True, size=64)
+    birth_date = fields.Date("Date of birth", required=True)
+    birth_city = fields.Many2one('res.city.it.code.distinct',
+                                 required=True,
+                                 string="City of birth")
+    birth_province = fields.Many2one('res.country.state',
+                                     required=True,
+                                     string="Province")
+    sex = fields.Selection([('M', "Male"), ('F', "Female")],
+                           required=True,
+                           string="Sex")
 
     @api.multi
     @api.onchange('birth_city')
     def onchange_birth_city(self):
         self.ensure_one()
-        res = {}
+
+        it = self.env.ref('base.it').id
+        res = {
+            'domain': {'birth_province': [('country_id', '=', it)]},
+            'value': {'birth_province': False}
+        }
+
         if self.birth_city:
-            ct = self.birth_city
-            res['domain'] = {
-                'birth_province': [('town_name', '=', ct.name)]
-            }
-        else:
-            res['domain'] = {'birth_province': []}
-        res['value'] = {'birth_province': ''}
+            # SMELLS: Add a foreign key in "res_city_it_code"
+            #          instead using the weak link "code" <-> "province".
+            #
+            city_ids = self.env['res.city.it.code'] \
+                .search([('name', '=', self.birth_city.name)])
+            provinces = city_ids.mapped('province')
+            province_ids = self.env['res.country.state'] \
+                .search([('country_id', '=', it), ('code', 'in', provinces)])
+
+            res['domain']['birth_province'] \
+                .append(('id', 'in', province_ids.ids))
+
+            if len(province_ids) == 1:
+                res['value']['birth_province'] = province_ids.id
+
+        if self.birth_province:
+            res.pop('value')
+
+        return res
+
+    @api.multi
+    @api.onchange('birth_province')
+    def onchange_birth_province(self):
+        self.ensure_one()
+
+        res = {'domain': {'birth_city': []}}
+
+        if not self.birth_city:
```

Comment:
Ciao, @eLBati!  
Perdona il ritardo... Me ne ero, bellamente, dimenticato. 😅

---

> Personalmente rimuoverei l'onchange `onchange_birth_province`, essendo sull'ultimo campo del form, la sua modifica non dovrebbe influire su campi precedenti.

Sono parzialmente d'accordo con la tua osservazione...  
Trovo corretto il fatto che non si debba influire su altri campi presenti nel form che potrebbero già essere valorizzati ma non che si debba rimuovere, del tutto, questa funzionalità.

Difatti, ecco il perché è presente di questa riga evidenziata.

---

Ho appena corretto il codice risolvendo l'errore che hai segnalato (mi ero perso una `s` in giro).  
Ora funziona. 😉